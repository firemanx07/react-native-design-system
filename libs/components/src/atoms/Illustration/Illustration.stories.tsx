import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import { StorybookScreen } from '../../storybook';
import {
  Illustration as IllustrationBase,
  PropsType as IllustrationProps,
  IllustrationSize,
  IllustrationName,
} from './Illustration';

const Story: ComponentMeta<typeof IllustrationBase> = {
  component: IllustrationBase,
  title: 'Atoms/Illustration',
};
export default Story;

const Template: ComponentStory<typeof IllustrationBase> = (
  args: IllustrationProps,
) => (
  <StorybookScreen title="Illustration:">
    <IllustrationBase {...args} />
  </StorybookScreen>
);

const parameters = {
  controls: {
    include: ['size', 'name'],
  },
};

export const Illustration = Template.bind({});
Illustration.parameters = parameters;
Illustration.args = {
  size: IllustrationSize.XL,
  name: IllustrationName.Board,
};
