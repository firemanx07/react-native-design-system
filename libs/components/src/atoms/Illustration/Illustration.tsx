import { styled } from '@proxym/themes';
import React, { memo } from 'react';
import { ImageProps } from 'react-native';

import * as illustrations from '../../assets/images/illustrations';

export enum IllustrationSize {
  // NOTE: it determines width of the image
  XL = 270,
  L = 200,
  M = 128,
  S = 60,
}

export enum IllustrationName {
  VideoPremium = 'VideoPremium',
  Video = 'Video',
  People = 'People',
  Upload = 'Upload',
  Dove = 'Dove',
  DoveUpdate = 'DoveUpdate',
  PolaroidMan = 'PolaroidMan',
  PolaroidWoman = 'PolaroidWoman',
}

export type PropsType = {
  name: IllustrationName;
  size?: IllustrationSize;
} & Omit<ImageProps, 'source'>;

export const Illustration = ({
  size = IllustrationSize.XL,
  name,
  ...imageProps
}: PropsType) => {
  if (!name) {
    return null;
  }

  return (
    <IllustrationImage
      {...imageProps}
      width={size}
      height={size * 0.8}
      source={illustrations[name]}
    />
  );
};

type IllustrationImageType = {
  height: number;
  width: number;
};

const IllustrationImage = styled.Image<IllustrationImageType>`
  width: ${({ width }) => width}px;
  height: ${({ height }) => height}px;
`;

export default memo(Illustration);
