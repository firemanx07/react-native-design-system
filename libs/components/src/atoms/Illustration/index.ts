export {
  default as Illustration,
  IllustrationSize,
  IllustrationName,
} from './Illustration';
