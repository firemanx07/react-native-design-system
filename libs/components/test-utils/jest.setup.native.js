import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
