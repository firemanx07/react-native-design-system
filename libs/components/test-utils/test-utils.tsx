import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
