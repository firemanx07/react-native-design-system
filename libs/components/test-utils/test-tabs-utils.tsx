import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
