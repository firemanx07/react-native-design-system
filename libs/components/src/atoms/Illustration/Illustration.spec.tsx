import { testAllStories } from '../../../test-utils';
import * as stories from './Illustration.stories';

describe('Illustration tests', () => {
  testAllStories(stories);
});
