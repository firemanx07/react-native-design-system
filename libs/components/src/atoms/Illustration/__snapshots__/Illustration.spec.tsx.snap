// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Illustration tests Illustration snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Rubik-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Illustration:
  </Text>
  <Image
    height={272}
    resizeMode="contain"
    source={
      {
        "uri": "",
      }
    }
    style={
      [
        {
          "height": 272,
          "width": 340,
        },
      ]
    }
    width={340}
  />
</View>
`;
